pkg/controller/daemonset/daemonset_controller.go

```diff
@@ -923,10 +923,19 @@ func (dsc *ReconcileDaemonSet) getNodesToDaemonPods(ds *appsv1alpha1.DaemonSet)
 		}
 		node := &corev1.Node{}
 		err = dsc.client.Get(context.TODO(), types.NamespacedName{Name: nodeName}, node)
-		if err == nil && CanNodeBeDeployed(node, ds) {
+		if err != nil {
+			if errors.IsNotFound(err) {
+				// Unable to find the target node in the cluster, which means it has been drained from cluster,
+				// we still add this node to nodeToDaemonPods in order that the daemon pods that failed to be
+				// scheduled on it will be removed in later reconcile loop.
+				nodeToDaemonPods[nodeName] = append(nodeToDaemonPods[nodeName], pod)
+			} else {
+				klog.V(4).Infof("get node %s failed: %v", nodeName, err)
+				continue
+			}
+		}
+		if CanNodeBeDeployed(node, ds) {
```

Comment:
If node is not found, the node object is empty. Will CanNodeBeDeployed return true in this case? If so, the Pods in the node are appended twice.

Reply:
update, PTAL